internal/command/command.go

```diff
@@ -0,0 +1,221 @@
+/*
+ * Flow CLI
+ *
+ * Copyright 2019-2021 Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package command
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/onflow/flow-cli/pkg/flow"
+	"github.com/onflow/flow-cli/pkg/flow/gateway"
+	"github.com/onflow/flow-cli/pkg/flow/services"
+	"github.com/onflow/flow-cli/pkg/flow/util"
+	"github.com/onflow/flow-go-sdk/client"
+	"github.com/psiemens/sconfig"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+type Command interface {
```

Comment:
I really like how you moved all this logic into the `command` package! 

After looking at the command implementations, I'd like to propose a slightly different way of structuring this that I think will allow you to use less code


```go
// -----
// command.go 

type RunCommand func (
  *cobra.Command, 
  []string,
  *flow.Project, 
  *services.Services,
) (Result, error)

type Command struct {
  Cmd *cobra.Command
  Config interface{}
  Run RunCommand
}

// -----
// contract-add.go

var AddContractCommand = &command.Command{
  Cmd: &cobra.Command{
    Use:     "add-contract <name> <filename>",
    Short:   "Deploy a new contract to an account",
    Example: `flow accounts add-contract FungibleToken ./FungibleToken.cdc`,
    Args:    cobra.ExactArgs(2),
  },
  Config: &flagsAddContract{},
  Run: func (
    cmd *cobra.Command,
    args []string,
    project *flow.Project,
    services *services.Services,
  ) (command.Result, error) {
    // ...
  }
}

Reply:
Awesome, thanks this is so much better :)